Editors/PathEditor.cpp

```diff
@@ -1,10 +1,238 @@
+    ui->roomView->mousePos = mousePos;
+    cursorPositionLabel->setText(tr("X %0, Y %1").arg(mousePos.x()).arg(mousePos.y()));
+    if (draggingPoint) {
+      pointsModel->setData(ui->roomView->selectedPointIndex, Path::Point::kXFieldNumber, ui->roomView->mousePos.x());
+      pointsModel->setData(ui->roomView->selectedPointIndex, Path::Point::kYFieldNumber, ui->roomView->mousePos.y());
+    } else
+      ui->pathPreviewBackground->update();  // manually call update to redraw cursor if no data changes
+  });
+
+  connect(ui->pathPreviewBackground, &AssetScrollAreaBackground::MousePressed, [=](Qt::MouseButton button) {
+    if (button == Qt::MouseButton::LeftButton) {
+      draggingPoint = true;
+      // Check for point at location (reverse loop to get last added one if dups)
+      for (int i = pointsModel->rowCount() - 1; i >= 0; i--) {
+        QPoint pt(pointsModel->data(i, Path::Point::kXFieldNumber).toInt(),
+                  pointsModel->data(i, Path::Point::kYFieldNumber).toInt());
+        if (pt == ui->roomView->mousePos) {
+          QModelIndex newSelectIndex = pointsModel->index(i, Path::Point::kXFieldNumber);
+          ui->pointsTableView->setCurrentIndex(newSelectIndex);
+          ui->pointsTableView->selectionModel()->select(newSelectIndex,
+                                                        QItemSelectionModel::QItemSelectionModel::ClearAndSelect);
+          return;
+        }
+      }
+      // No point found. Add one and select it
+      InsertPoint(pointsModel->rowCount(), ui->roomView->mousePos.x(), ui->roomView->mousePos.y(),
+                  ui->speedSpinBox->value());
+    }
+  });
+
+  connect(ui->pathPreviewBackground, &AssetScrollAreaBackground::MouseReleased, [=](Qt::MouseButton button) {
+    if (button == Qt::MouseButton::LeftButton) {
+      draggingPoint = false;
+    }
+  });
+
+  connect(ui->pointsTableView->selectionModel(), &QItemSelectionModel::selectionChanged, [this]() {
+    int selectIndex = -1;
+    if (ui->pointsTableView->selectionModel()->hasSelection())
+      selectIndex = ui->pointsTableView->selectionModel()->currentIndex().row();
+    ui->roomView->selectedPointIndex = selectIndex;
+    ui->pathPreviewBackground->update();
+    if (selectIndex != -1) ui->deletePointButton->setDisabled(false);
+  });
+
+  ui->deletePointButton->setDisabled(true);
+
+  BaseEditor::RebindSubModels();
+}
+
+bool PathEditor::eventFilter(QObject* obj, QEvent* event) {
+  if (pointsModel != nullptr) {
+    // Resize columns to view size
+    if (obj == ui->pointsTableView && event->type() == QEvent::Resize) {
+      QResizeEvent* resizeEvent = static_cast<QResizeEvent*>(event);
+      int cc = pointsModel->columnCount() - 1;
+      for (int c = 1; c < cc; ++c) {  // column 1 is hidden
+        ui->pointsTableView->setColumnWidth(c, (resizeEvent->size().width()) / cc);
+      }
+    }
+  }
+  return QWidget::eventFilter(obj, event);

```

Comment:
Not familiar enough with Qt to know if this is always supposed to be called. You're handling some events above. Is this what invokes the base view's event handler?

Reply:
Event filters capture events from any qobject you give a pointer to with installeventfiltet(obj) I'm only capturing the tableviews events here so only those are handled here. Qt has no option to equally stretch columns it can only stretch the last so this is only way to keep even columns other than making our own tableviews inherited and installing an eventfilter there instead

Reply:
Event filters capture events from any qobject you give a pointer to with installeventfiltet(obj) I'm only capturing the tableviews events here so only those are handled here. Qt has no option to equally stretch columns it can only stretch the last so this is only way to keep even columns other than making our own tableviews inherited and installing an eventfilter there instead

Reply:
I should clarify: I get why we call it in general, but I'm not sure we should be calling it when we've already handled the event. But assuming you're seeing the right behavior for events we *are* handling here, this should be fine.